api/src/main/java/com/epam/pipeline/manager/datastorage/providers/gcp/GSBucketStorageHelper.java

```diff
@@ -473,4 +544,53 @@ private void deleteBucket(final String bucketName, final Storage client) {
     private String trimRegionZone(final String region) {
         return StringUtils.substring(region, 0, REGION_ZONE_LENGTH);
     }
+
+    private void disableLifecycleRulesIfNeeded(final Storage client, final String bucketName,
+                                               final StoragePolicy policy) {
+        if (Objects.nonNull(policy.getLongTermStorageDuration())
+                || Objects.nonNull(policy.getShortTermStorageDuration())
+                || Objects.nonNull(policy.getBackupDuration())) {
+            return;
+        }
+        try {
+            storageRestApiClient = buildGoogleStorageRestApiClient();
+            final GoogleCredentials credentials = (GoogleCredentials) client.getOptions().getScopedCredentials();
+            final String token = credentials.refreshAccessToken().getTokenValue();
+            executeRequest(() ->
+                    storageRestApiClient.disableLifecycleRules(buildToken(token), LIFECYCLE_CONTENT_TYPE,
+                            bucketName, LIFECYCLE_FIELD, EMPTY_LIFECYCLE_CONTENT));
+        } catch (IOException e) {
+            throw new DataStorageException(String
+                    .format("Failed to disable lifecycle rules for bucket %s", bucketName), e);
+        }
+    }
+
+    private <T> T executeRequest(Supplier<Call<T>> request) {
```

Comment:
I'd suggest to move all logic for creating client and executing request to `GoogleStorageRestApiClient` to a separate class

Reply:
done